shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -175,14 +173,11 @@ - (void)setViewController:(FlutterViewController*)viewController {
   _viewController = [viewController getWeakPtr];
   self.iosPlatformView->SetOwnerViewController(_viewController);
   [self maybeSetupPlatformViewChannels];
+  [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(onFlutterVCWillDeallocNotif:) name:FlutterViewControllerWillDealloc object:viewController];
+}
 
-  self.flutterViewControllerWillDeallocObserver =
-      [[NSNotificationCenter defaultCenter] addObserverForName:FlutterViewControllerWillDealloc
-                                                        object:viewController
-                                                         queue:[NSOperationQueue mainQueue]
-                                                    usingBlock:^(NSNotification* note) {
-                                                      [self notifyViewControllerDeallocated];
-                                                    }];
+- (void)onFlutterVCWillDeallocNotif:(NSNotification *)aNotification {
+  [self notifyViewControllerDeallocated];
```

Comment:
Is there a reason to have this new method at all?  Can you add a notification parameter to notifyViewControllerDeallocated (assuming this isn't exposed in a header, I'm not looking at the code at the moment)
```objective-c
[[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(notifyViewControllerDeallocated:) name:FlutterViewControllerWillDealloc object:viewController];
```
```objective-c
- (void)notifyViewControllerDeallocated:(NSNotification *)notification {
```

Also, does this still need to dispatch to the mainQueue if this isn't the main queue?  There's no guarantee about what queue the notification will be posted on.